Add s3 acl documentation (#118)
syseleven/grav-docs-syseleven-stack: user/pages/04.Reference/05.object-storage/docs.en.md

bucket listing                   | yes       |
input compression                | yes       |
request signature authentication | yes       |

> bucket/object ACL                | yes*      |

Author: I'd rather leave "partially" as is raises fewer expectations.

Reviewer: Reset the value to partially, fine for me =)

access logging                   | no        |
website hosting configuration    | no        |

> For setting up bucket/object ACLs we suggest to use [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html). We have prepared a [guide in our Tutorials section](../../02.Tutorials/11.object-storage-acls/docs.en.md).

Reviewer 2: This change removes the explanation of "partial", but leaves the word "partial" in the table above. As I don't feel like taking the responsability for full ACL support, so we should not necessarily change it to a "yes", but then keep an explanation (although different) for what we mean by "partial". For example, it seems that access to the index cannot be restricted by ACL, as in AWS and even if this unexpectedly works after latest updates, there will likely remain differences.

Reviewer: Good point, lets have another closer look what AWS offers for features. I am not sure if the directory listing part is more a topic of the bucket policies (see the ACL part of their docs : https://docs.aws.amazon.com/AmazonS3/latest/dev-retired/acl-overview.html). It may even be achieved by narrowing down ACLs to WRITE only (depending if that is practicable for the customer use-case) 

We should refer to the canned ACLs as the implementation as not all of these predefined ACLs even exists in our implementation.

Reviewer: Added a further note to make clear we do not offer the complete Amazon S3 feature set (I wish we would tho).

Reviewer: Issue handled in another comment! =)

access logging                   | no        |
website hosting configuration    | no        |

> We do not offer the complete featureset compared to Amazon S3. Thus we cannot guarentee the same user experience. Certain API calls may fail and receive a `501 Not Implemented` response. If you are in doubt, feel free to contact our [Cloud-Support (cloudsupport@syseleven.de)](../../06.Support/default.en.md).

Reviewer 3: ```suggestion
We do not offer the complete featureset compared to Amazon S3. Thus we cannot guarantee the same user experience. Certain API calls may fail and receive a `501 Not Implemented` response. If you are in doubt, feel free to contact our [Cloud-Support (cloudsupport@syseleven.de)](../../06.Support/default.en.md).
```

> We do not offer the complete featureset compared to Amazon S3. Thus we cannot guarantee the same user experience. Certain API calls may fail and receive a `501 Not Implemented` response. If you are in doubt, feel free to contact our [Cloud-Support (cloudsupport@syseleven.de)](../../06.Support/default.en.md).

Reviewer 3: ```suggestion
We do not offer the complete feature set compared to Amazon S3. Thus we cannot guarantee the same user experience. Certain API calls may fail and receive a `501 Not Implemented` response. If you are in doubt, feel free to contact our [Cloud-Support (cloudsupport@syseleven.de)](../../06.Support/default.en.md).
```

> For setting up bucket/object ACLs we suggest to use [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html). Not all predefined (canned) ACLs of [AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html) are supported. We are supporting the following canned ACLs: `private`,`public-read`,`public-read-write`,`authenticated-read`. Not supported canned ACLs will be interpreted as the default `private` ACL. We have prepared a [guide in our Tutorials section](../../02.Tutorials/11.object-storage-acls/docs.en.md) which shows how to setup custom ACLs.

Reviewer 3: ```suggestion
For setting up bucket/object ACLs we suggest to use [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html). Not all predefined (canned) ACLs of [AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html) are supported. We are supporting the following canned ACLs: `private`,`public-read`,`public-read-write`,`authenticated-read`. Not supported canned ACLs will be interpreted as the default `private` ACL. We have prepared a [guide in our Tutorials section](../../02.Tutorials/11.object-storage-acls/docs.en.md) which shows how to set up custom ACLs.
```

> For setting up bucket/object ACLs we suggest to use [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html). Not all predefined (canned) ACLs of [AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html) are supported. We are supporting the following canned ACLs: `private`,`public-read`,`public-read-write`,`authenticated-read`. Not supported canned ACLs will be interpreted as the default `private` ACL. We have prepared a [guide in our Tutorials section](../../02.Tutorials/11.object-storage-acls/docs.en.md) whichs shows how to setup custom ACLs.

Reviewer 3: ```suggestion
For setting up bucket/object ACLs we suggest to use [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html). Not all predefined (canned) ACLs of [AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html) are supported. We are supporting the following canned ACLs: `private`,`public-read`,`public-read-write`,`authenticated-read`. Not supported canned ACLs will be interpreted as the default `private` ACL. We have prepared a [guide in our Tutorials section](../../02.Tutorials/11.object-storage-acls/docs.en.md) which shows how to setup custom ACLs.
```

> For setting up bucket/object ACLs we suggest to use [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html). Not all predefined (canned) ACLs of [AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html) are supported. We are supporting the following canned ACLs : `private`,`public-read`,`public-read-write`,`authenticated-read`. Not supported canned ACLs will be interpreted as the default `private` ACL. We have prepared a [guide in our Tutorials section](../../02.Tutorials/11.object-storage-acls/docs.en.md) whichs shows how to setup custom ACLs.

Reviewer 3: ```suggestion
For setting up bucket/object ACLs we suggest to use [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html). Not all predefined (canned) ACLs of [AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html) are supported. We are supporting the following canned ACLs: `private`,`public-read`,`public-read-write`,`authenticated-read`. Not supported canned ACLs will be interpreted as the default `private` ACL. We have prepared a [guide in our Tutorials section](../../02.Tutorials/11.object-storage-acls/docs.en.md) whichs shows how to setup custom ACLs.
```

You can use these URLs to refer to the uploaded files as static assets in your web applications.

> If you plan to use more complex ACLs we suggest you checkout the boto3 client.

Reviewer 2: This line looks like a bridge passage but imo rather belongs to the boto3 section, than to the s3cmd section. I suggest to move it behind the section headline.

Reviewer: Alright! After we have fully reworked the ACL tutorial we may also rework the background article as we might have duplicate information between both articles

Reviewer: The line was already removed and the boto3 sections contains the hint it would be tool of choice for more complex ACL scenarios

You can use these URLs to refer to the uploaded files as static assets in your web applications.

### Boto3

> Information about the `boto3` python S3 library can be found [here](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html). We suggest to make use of this library to setup and manage more complex ACLs.

Reviewer 3: ```suggestion
Information about the `boto3` python S3 library can be found [here](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html). We suggest to make use of this library to set up and manage more complex ACLs.
```

Using the following python snippet we can configure our client:

```python
import boto3
import botocore
# Get our session
session = boto3.session.Session()
s3 = session.resource(

>    service_name = 's3',

Reviewer 3: indentation should be 4 spaces

Reviewer: fixed in latest commit

   aws_access_key_id = "< REPLACE ME >",
   aws_secret_access_key = "< REPLACE ME >",
   endpoint_url = 'https://s3.dbl.cloud.syseleven.net'

>    )

Reviewer 3: ```suggestion
)
```

# Get our client

> s3client=s3.meta.client

Reviewer 3: ```suggestion
s3client = s3.meta.client
```

```

For (re)creating a bucket we can use following snippet:

```python
bucket = "myBucket"
try:
    s3client.create_bucket(Bucket=bucket)

> except:

Reviewer 3: ```suggestion
except Exception:
```

   print("There was a problem creating the bucket '{}', it may have already existed".format(bucket))
   oldbucket = s3.Bucket(bucket)
   # Cleanup remaining objects in bucket
   oldbucket.objects.all().delete()
   s3client.delete_bucket(Bucket=bucket)
   s3client.create_bucket(Bucket=bucket)
```

To upload our first objects we may use following commands:

```python
bucket = "myBucket"

> s3client.put_object(Body="secret",Bucket=bucket,Key="private-scope-object")

Reviewer 3: ```suggestion
s3client.put_object(Body="secret", Bucket=bucket, Key="private-scope-object")
```
